src/hash_table.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+from typing import Union
+import random
+
+
+class Node:
```

Comment:
Ideally, `Node` should be a private class inside `ChainHashTable`.

Reply:
Make sense. I can do it like this, but isn't it verbose? @toet251 do you have any idea?
```
class ChainHashTable:
    class Node:
        def __init__(self, key: str, val: Union[str, float], next: ChainHashTable.Node = None):
            self.key = key
            self.val = val
            self.next = next

    def __init__(self, n: int):
        self.__bucket = [ChainHashTable.Node(None, None)] * n
```

Reply:
Idk. Maybe it is the only way for private class. Another accessing way is self.Node(), shorter anyway :))